Fix Debug Cheats Tools

iamc24 opened the pull request:
- Fix functionality on browsers
- Fix loading held items (credit to @ajthompson's solution in #403)
- Fix loading the Pokédex
- Fix name filtering in the Pokédex so that it is no longer case-sensitive
- Fix region filtering in the Pokédex so that it filters only for a Pokémon's native region instead of also allowing the region names in the route data used for hints to be in what the filter checks
- Add missing "None" region used for Meltan line to region filtering
- Fix Meltan line showing in Pokédex before Alola
- Fix currency cheat to allow removing currency with negative numbers in input (#425 & #427)

iamc24 committed:
Update debugcheatstools.user.js

Fix loading held items and pokedex; Fix region filtering in pokedex

iamc24 committed:
Update debugcheatstools.user.js

Better fix for region filtering

iamc24 commented:
Working on improving hints in the Pokédex so most methods of obtaining are covered instead of just routes

iamc24 committed:
Update debugcheatstools.user.js

Fix Pokédex name filtering being case-sensitive; Fix none being default region filtering option

iamc24 committed:
Update debugcheatstools.user.js

Fix currency cheat to allow removing currency with negative numbers in input

Memally35972 commented:
it doesn't work with pokedollars

iamc24 committed:
Update debugcheatstools.user.js

Improve currency handling

Memally35972 commented:
plsease also add underground rewards

iamc24 committed:
Update debugcheatstools.user.js

Fix Meltan line showing in Pokédex before Alola

iamc24 committed:
Update debugcheatstools.user.js (#1)

Refactor to make script work in browser; Bump version; Add credits

Memally35972 commented:
Origin forms , Noble forms , Hisuian forms , and the four exclusive paldea wanders @iamc24

iamc24 commented:
> Origin forms , Noble forms , Hisuian forms , and the four exclusive paldea wanders @iamc24

Those would all be under Hisui or Paldea in the region selector. Since those regions aren't in the game yet, they won't get loaded for the Pokédex. Like I said before, this is to prevent causing issues with regions you haven't been to yet, and I won't be changing it. You are free to modify it yourself to remove the region restriction.

iamc24 committed:
Update debugcheatstools.user.js

Slightly simplify event handler logic for gems, pokeballs, berries;

iamc24 committed:
Update debugcheatstools.user.js

fix indents

iamc24 commented:
> They are in the game (noble , Husain, Origin ) 🙃 [Pokécliker](https://dev.pokeclicker.com/)

Didn't know about the dev site; I have only been using the regular build. Tested on the dev build and they all load for me on it, so I wouldn't know why they aren't for you.

> please make it if you write (-0) it automatically removes all of your money until to $100, because for whatever reason it doesn’t change for me(my money) if I buy stuff?
> 
> PS:My current money is $365,625,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000, 000,000,000,000,000,000,000/ (3.65625e+260T)

Tested with this amount; Just put `-365625000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000` in for input

iamc24 commented:
> plsease also add underground rewards

@Memally35972 Added a branch to my repository that adds underground and other misc items

Memally35972 commented:
> Added a branch to my repository that adds underground and other misc items

where

Memally35972 commented:
> > They are in the game (noble , Husain, Origin ) 🙃 [Pokécliker](https://dev.pokeclicker.com/)
> 
> Didn't know about the dev site; I have only been using the regular build. Tested on the dev build and they all load for me on it, so I wouldn't know why they aren't for you.
dont know how to put it in websites only desktop😅

Memally35972 commented:
> > Origin forms , Noble forms , Hisuian forms , and the four exclusive paldea wanders @iamc24
> 
> You are free to modify it yourself to remove the region restriction at your own risk.
how

iamc24 committed:
Update debugcheatstools.user.js

Fix issue with duplicate event handlers being created, multiplying the amount of currency/items added/removed

iamc24 committed:
Update debugcheatstools.user.js

Slightly simplify more event handlers; Fix evolution item name capitalization; Capitalize held items tab name; Improve consistency between ids and comments for evolution/held items

iamc24 committed:
Update debugcheatstools.user.js

Fix indentation

iamc24 committed:
Update debugcheatstools.user.js

Re-add custom html attribute accidentally removed when simplifying event handlers; Fix minor indentation issue

iamc24 committed:
Update debugcheatstools.user.js

Fix variable typo

Memally35972 commented:
for the scroll
